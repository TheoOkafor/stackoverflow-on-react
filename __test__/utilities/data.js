const data = {
  answer: {
    accepted: null,
    body: 'What?',
    comments: [
      {
        answerid: 17,
        body: 'Jonzing',
        id: 30,
        questionid: 4,
        timesubmitted: '2018-11-21T12:35:08.000Z',
        userid: 4,
        username: 'teddy01'
      },
    ],
    downvotes: 0,
    id: 74,
    questionid: 38,
    timesubmitted: '2018-11-21T18:20:11.000Z',
    upvotes: 0,
    userid: 4,
    username: 'teddy01',
    votesCount: 0,
  },
  comment: {
    answerid: 17,
    body: 'Jonzing',
    id: 30,
    questionid: 4,
    timesubmitted: '2018-11-21T12:35:08.000Z',
    userid: 4,
    username: 'teddy01'
  },
  sampleAnswer: {
    data: {
      body: 'Jonzing',
      questionid: 4,
      timesubmitted: '2018-11-21T12:35:08.000Z',
      userid: 4,
      username: 'teddy01'
    },
  },
  question: {
    data: {
      answers: [
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 4,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 1,
          userid: 3,
          username: 'chanpion',
          votesCount: 1
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 8,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 12,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [],
          downvotes: 0,
          id: 16,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: false,
          body: 'Lorem ipsum dolor sit amet, ',
          comments: [{
            answerid: 17,
            body: 'Jonzing',
            id: 30,
            questionid: 4,
            timesubmitted: '2018-11-21T12:35:08.000Z',
            userid: 4,
            username: 'teddy01'
          }],
          downvotes: 0,
          id: 17,
          questionid: 4,
          timesubmitted: '2018-08-22T11:13:09.000Z',
          upvotes: 0,
          userid: 3,
          username: 'chanpion',
          votesCount: 0
        },
        {
          accepted: null,
          body: 'worddy',
          comments: [],
          downvotes: 0,
          id: 67,
          questionid: 4,
          timesubmitted: '2018-11-21T12:23:12.000Z',
          upvotes: 0,
          userid: 4,
          username: 'teddy01',
          votesCount: 0
        }
      ],
      body: 'Lorem ipsum dolor sit amet, ',
      id: 4,
      timesubmitted: '2018-08-22T11:13:09.000Z',
      title: 'How can one learn it all so fast?',
      userid: 3,
      username: 'chanpion'
    },
    message: 'Question 4 found',
    statusCode: 200
  },
  questions: [
    {
      id: 1,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
    {
      id: 2,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
    {
      id: 3,
      title: 'why do yo shake?',
      body: 'Lorem ipsum dolor sit amet, ',
      userid: 3,
      username: 'chanpion',
      answers: [],
    },
  ],
  newQuestion: {
    id: 1,
    title: 'why do yo shake?',
    body: 'Lorem ipsum dolor sit amet, ',
    userid: 3,
    username: 'chanpion',
    answers: [],
  },
  payload: {
    username: 'Xtreme',
    'x-access-token': 'token',
  },
  userData: {
    firstName: 'Xtreme',
    lastName: 'Prodigy',
    username: 'Xtreme',
    password: 'password',
    confirmpassword: 'password',
  },
};

export default data;
