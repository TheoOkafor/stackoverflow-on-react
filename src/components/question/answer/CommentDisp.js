import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CommentDisp = (props) => {
  return (
    <p className="comment-display">
      <small>{props.comment.body}
        <span className="right">
          <Link to= {`/users/${props.comment.userid}`}
            className="inherit">
            <b>{props.comment.username}</b>
          </Link>
          ({new Date(props.comment.timesubmitted).toDateString()})
        </span>
      </small>
    </p>
  );
};

CommentDisp.propTypes = {
  comment: PropTypes.object,
};

export default CommentDisp;
