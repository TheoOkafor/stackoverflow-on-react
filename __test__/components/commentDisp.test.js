import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import CommentDisp from '../../src/components/question/answer/CommentDisp';
import data from '../utilities/data';

test('Test that the CommentDisp displays', () => {
  const component = renderer.create(
    <MemoryRouter>
      <CommentDisp comment={data.comment} index={12}/>
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});
