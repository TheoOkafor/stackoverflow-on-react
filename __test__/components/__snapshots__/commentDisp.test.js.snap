// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the CommentDisp displays 1`] = `
<p
  className="comment-display"
>
  <small>
    Jonzing
    <span
      className="right"
    >
      <a
        className="inherit"
        href="/users/4"
        onClick={[Function]}
      >
        <b>
          teddy01
        </b>
      </a>
      (
      Wed Nov 21 2018
      )
    </span>
  </small>
</p>
`;
